io/include/pcl/io/grabber.h

```diff
@@ -242,19 +242,30 @@ namespace pcl
   template<typename T> boost::signals2::signal<T>*
   Grabber::createSignal ()
   {
-    if (providesCallback<T> ())
+    using Signal = boost::signals2::signal<T>;
+    using Base = boost::signals2::signal_base;
+    struct DefferedPtr {
+      operator std::unique_ptr<Base>() const { return std::make_unique<Signal>(); }
+    };
```

Comment:
You might want to add an explanation here why this specific construct is needed.